pkg/cmd/pulumi/new.go

```diff
@@ -390,6 +451,205 @@ func useSpecifiedDir(dir string) (string, error) {
 	return cwd, nil
 }
 
+type AIPromptRequestBody struct {
```

Comment:
Nit: I do wonder if we should factor some of the AI-specific utilities out of `new.go` to a separate file.

Also, does this struct need to be exported?

Reply:
I wondered the same thing here - I'm perfectly happy to pull all the AI stuff into a separate file here, but wasn't sure what the desired pattern was. I'll get that stuff moved now (it doesn't need to be exported, no).